txcache/selectionTracker.go

```diff
@@ -140,48 +119,54 @@ func (st *selectionTracker) checkReceivedBlockNoLock(blockBody *block.Body, bloc
 	hasNewTransactions := len(blockBody.MiniBlocks) != 0
 	hasNoNewExecutionResults := len(blockHeader.GetExecutionResultsHandlers()) == 0
 
+	// should receive empty block or a block with new execution results
 	if hasNewTransactions && hasNoNewExecutionResults {
-		log.Warn("selectionTracker.checkReceivedBlockNoLock: received bad block while max tracked blocks is reached. "+
-			"should receive empty block or a block with new execution results",
+		log.Warn("selectionTracker.checkReceivedBlockNoLock: received non-tolerated block while max tracked blocks is reached. "+
 			"len(st.blocks)", len(st.blocks),
 		)
 
 		return errBadBlockWhileMaxTrackedBlocksReached
 	}
 
+	// received an empty block or a block with new execution results
 	log.Warn("selectionTracker.checkReceivedBlockNoLock: max tracked blocks reached "+
-		"but received a tolerated block - an empty block or a block with new execution results",
-		"len(st.blocks)", len(st.blocks))
+		"but received a tolerated block",
+		"len(st.blocks)", len(st.blocks),
+		"nonce", blockHeader.GetNonce())
 
 	return nil
 }
 
-func (st *selectionTracker) validateTrackedBlocks(
+// validateTrackedBlocksAndCompileBreadcrumbs is used when a new block is proposed.
```

Comment:
Allright, explicit :heavy_check_mark: 

(does more things, yes, but that's due to a flow optimization).